src/iomap.cpp

```diff
@@ -69,85 +69,90 @@ Tile* IOMap::createTile(Item*& ground, Item* item, uint16_t x, uint16_t y, uint8
 bool IOMap::loadMap(Map* map, const std::string& fileName)
 {
 	int64_t start = OTSYS_TIME();
-	OTB::Loader loader{fileName, OTB::Identifier{{'O', 'T', 'B', 'M'}}};
-	auto& root = loader.parseTree();
+	try {
```

Comment:
I don't like the scope of this `try` block, it catches a lot more errors than it should, e.g. if something throws during tile parsing, we may end up in an inconsistent state.

Reply:
Well I wanted to catch only the constructor, but I couldn't because of the variable being enclosed in try catch block, thus rest of the code wont proceed. I could probably use pointer for this though.

Reply:
Hmm, this is actually a tough one, because parsing props may still cause us to end up in a partially-loaded state. I usually prefer to make such operations "transactional" (either succeeds completely or has no effect), but I guess it's way out of scope of this PR. 

I think it's sufficient to just change the type you accept in `catch` to `catch (const InvalidOTBFormat& err)`

Reply:
Honestly gonna sound like a noob here, but I tried but couldn't make it to work. No matter what I did it wouldn't find the custom exception type. I included fileloader.h on the top and still wasn't able to find it somehow...

Reply:
`const OTB::LoadError&` should work just fine.